src/commands/Tags/tag.ts

```diff
@@ -1,59 +1,31 @@
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
- */
+// Copyright (c) 2018 BDISTIN. All rights reserved. MIT license.
+// Source: https://github.com/KlasaCommunityPlugins/tags
 
 import { SkyraCommand } from '@lib/structures/SkyraCommand';
 import { GuildSettings } from '@lib/types/settings/GuildSettings';
-import { cutText } from '@utils/util';
-import { CommandStore, KlasaMessage } from 'klasa';
-
+import { cutText, getColor } from '@utils/util';
+import { KlasaMessage, CommandOptions } from 'klasa';
+import { UserRichDisplay } from '@lib/structures/UserRichDisplay';
+import { chunk } from '@klasa/utils';
+import { BrandingColors } from '@utils/constants';
+import { MessageEmbed } from 'discord.js';
+import { ApplyOptions, requiresPermission } from '@skyra/decorators';
+
+@ApplyOptions<CommandOptions>({
+	aliases: ['tags'],
+	description: language => language.tget('COMMAND_TAG_DESCRIPTION'),
+	extendedHelp: language => language.tget('COMMAND_TAG_EXTENDED'),
+	runIn: ['text'],
+	subcommands: true,
+	requiredPermissions: ['EMBED_LINKS', 'MANAGE_MESSAGES'],
+	usage: '<add|remove|edit|source|list|show:default> (tag:tagname) [content:...string]',
+	usageDelim: ' '
+})
 export default class extends SkyraCommand {
 
-	public constructor(store: CommandStore, file: string[], directory: string) {
-		super(store, file, directory, {
-			description: language => language.tget('COMMAND_TAG_DESCRIPTION'),
-			extendedHelp: language => language.tget('COMMAND_TAG_EXTENDED'),
-			runIn: ['text'],
-			subcommands: true,
-			usage: '<add|remove|edit|source|list|show:default> (tag:tagname) [content:...string]',
-			usageDelim: ' '
-		});
-
-		this.createCustomResolver('tagname', (arg, possible, message, [action]) => {
-			if (action === 'list') return undefined;
-			if (!arg) throw message.language.tget('RESOLVER_INVALID_STRING', possible.name);
-			if (arg.includes('`') || arg.includes('\u200B')) throw message.language.tget('COMMAND_TAG_NAME_NOTALLOWED');
-			if (arg.length > 50) throw message.language.tget('COMMAND_TAG_NAME_TOOLONG');
-			return arg.toLowerCase();
-		});
-	}
-
+	@requiresPermission(4, (message: KlasaMessage) => { throw message.language.tget('COMMAND_TAG_PERMISSIONLEVEL'); })
```

Comment:
Add 4 to the PermissionLevels enum (`src/lib/types/Enums.d.ts`) and reference it here and other places.